Having an import issue

```
      assert_allclose(demeaned, expected)
        ^^^^^^^^^^^^^^^
E       NameError: name 'assert_allclose' is not defined
```

--- a/statsmodels/tools/grouputils.py
+++ b/statsmodels/tools/grouputils.py
@@ -98,7 +98,7 @@ def group_sums(x, group, use_bincount=True):
     -----
     Both code paths previously disagreed on orientation (``use_bincount``
     returned ``(n_features, n_groups)``). They now both return
-    ``(n_groups, n_features)`` so indexing by group id works (#9921).
+    ``(n_groups, n_features)`` so indexing by group id works (GH9921).
     """
     x = np.asarray(x)
     group = np.asarray(group).squeeze()

--- a/statsmodels/tools/grouputils.py
+++ b/statsmodels/tools/grouputils.py
@@ -82,11 +82,16 @@ def group_sums(x, group, use_bincount=True):
     Parameters
     ----------
     x : array_like
-        Data of shape ``(nobs,)`` or ``(nobs, n_features)``.
-    group : array_like, integer
-        Group labels of shape ``(nobs,)``.
+        Data of shape ``(nobs,)`` or ``(nobs, n_features)``. Higher-dimensional
+        ``x`` is only supported when ``use_bincount=False``.
+    group : array_like of non-negative int
+        Non-negative integer group labels of shape ``(nobs,)``. For predictable
+        indexing (e.g. ``result[group]``), groups should be coded as
+        ``0, 1, ..., n_groups-1``.
     use_bincount : bool, default True
         Use ``np.bincount`` when True, otherwise a pure-Python group loop.
+        The bincount path expects non-negative, reasonably consecutive codes
+        and may re-label via ``pd.factorize`` when ``max(group)`` is large.
 
     Returns
     -------
@@ -271,7 +276,8 @@ class Group:
         sums_g = group_sums(x, self.group_int, use_bincount=use_bincount)
         # group_sums returns (n_groups, n_features)
         counts = np.bincount(self.group_int)
-        # avoid divide-by-zero for empty labels
+        # Defensive: bincount may include zero-count bins for sparse codes.
+        # group_int from combine_indices should not contain empty labels.
         counts = np.maximum(counts, 1)
         means_g = sums_g / counts[:, None]
         x_demeaned = x - means_g[self.group_int]

--- a/statsmodels/tools/tests/test_grouputils.py
+++ b/statsmodels/tools/tests/test_grouputils.py
@@ -1,7 +1,7 @@
 from statsmodels.compat.pandas import assert_frame_equal, assert_series_equal
 
 import numpy as np
-from numpy.testing import assert_equal
+from numpy.testing import assert_allclose, assert_equal
 import pandas as pd
 import pytest
 from scipy import sparse
@@ -732,7 +732,7 @@ def test_group_sums():
 
 
 def test_group_sums_orientation_consistent():
-    """Both use_bincount paths return (n_groups, n_features) (#9921)."""
+    """Both use_bincount paths return (n_groups, n_features) (GH9921)."""
     g = np.array([0, 0, 1, 2, 1, 1, 2, 0])
     x = np.arange(len(g) * 3).reshape(len(g), 3, order="F")
     a = group_sums(x, g, use_bincount=True)
@@ -742,7 +742,7 @@ def test_group_sums_orientation_consistent():
 
 
 def test_group_demean_default():
-    """group_demean works with default use_bincount and demeans by group size (#9921)."""
+    """group_demean works with default use_bincount and demeans by group size (GH9921)."""
     g = np.array([0, 0, 1, 1, 1, 2, 2, 2])
     x = np.array([1.0, 3.0, 1.0, 2.0, 3.0, 10.0, 20.0, 30.0])
     demeaned, means = Group(g).group_demean(x)

